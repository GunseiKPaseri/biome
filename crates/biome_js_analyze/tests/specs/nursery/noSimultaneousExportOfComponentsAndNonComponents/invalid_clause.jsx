const SampleComponent = () => <></>
const test = 100
function hoge () {
  return 100
}
const piyo = 100

export {SampleComponent, test, hoge, piyo as fuga}
