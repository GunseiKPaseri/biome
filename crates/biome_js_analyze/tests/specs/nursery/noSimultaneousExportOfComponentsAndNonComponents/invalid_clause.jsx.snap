---
source: crates/biome_js_analyze/tests/spec_tests.rs
expression: invalid_clause.jsx
---
# Input
```jsx
const SampleComponent = () => <></>
const test = 100
function hoge () {
  return 100
}
const piyo = 100

export {SampleComponent, test, hoge, piyo as fuga}

```

# Diagnostics
```
invalid_clause.jsx:8:26 lint/nursery/noSimultaneousExportOfComponentsAndNonComponents ━━━━━━━━━━━━━━

  ! Components and non-components functions are exported at the same time.
  
    6 │ const piyo = 100
    7 │ 
  > 8 │ export {SampleComponent, test, hoge, piyo as fuga}
      │                          ^^^^
    9 │ 
  
  i React components and regular functions must be exported in separate files.
  

```

```
invalid_clause.jsx:8:32 lint/nursery/noSimultaneousExportOfComponentsAndNonComponents ━━━━━━━━━━━━━━

  ! Components and non-components functions are exported at the same time.
  
    6 │ const piyo = 100
    7 │ 
  > 8 │ export {SampleComponent, test, hoge, piyo as fuga}
      │                                ^^^^
    9 │ 
  
  i React components and regular functions must be exported in separate files.
  

```

```
invalid_clause.jsx:8:46 lint/nursery/noSimultaneousExportOfComponentsAndNonComponents ━━━━━━━━━━━━━━

  ! Components and non-components functions are exported at the same time.
  
    6 │ const piyo = 100
    7 │ 
  > 8 │ export {SampleComponent, test, hoge, piyo as fuga}
      │                                              ^^^^
    9 │ 
  
  i React components and regular functions must be exported in separate files.
  

```
